meetings/cranelift/2022/cranelift-03-07.md

```diff
@@ -16,5 +16,62 @@
+    - fitzgen: Are these always applied to extractors or constructors
+    - avanhatt: Applied to both
+    - cfallin: I would expect it to be the same for extractors or constructors, it aligns with the goals of rewriting engines
+    - fitzgen: Re not repeating ourselves: I think it makes sense to infer types from ISLE parameter types
+    - cfallin: I would refine that and say: 1 place and not 0 places. There should be a declaration somewhere
+    - fitzgen: Right, I think on the type declarations
+    - bjorne3: It would be nice if this was in ISLE instead of comments
+    - avanhatt: At what point do we merge this prototype? Maybe it makes sense then to transition from special comments to real ISLE syntax
+    - cfallin: Feel free to hack up our parser in your branch, and we could take a change for general annotation syntax out of it too.
+    - fitzgen: I dunno how much we want to really copy that, I find it a bit over-engineered. But we could have arbitrary S-expression tags
+    - cfallin: I had a question about fallibility on the conceptual level. Is it that : `assumption` encodes the postcondition if the match succeeds? Or is it encoding the condition for the match to succeed at all? I’d expect the first
+    - avanhatt: Yes, the first.
+    - cfallin: One thing that would be nice to reason about with fallibility is coverage, knowing that our conditions actually cover all cases. Is there a plan to have that?
+    - avanhatt: Set up for it and have talked about it, no actual plan for it. Dovetails with multi rule reasoning
+    - cfallin: Is there a mechanism for abstraction? “Helper functions” for similar operations
+    - avanhatt: Yeah, I think we should. In particular there’s lots of noops. I think we should, but there’s no concrete proposal for it yet.
+    - fitzgen: I think we talked about this on Zulip, but the representation for bitwidth is the same on both left and right hand sides? So we don’t have the ability to represent undefined bits in eg lowering registered width
+    - avanhatt: Would be cool to do, aren’t doing it yet, can extend to do it.
+    - cfallin: Would prefer to not special case things, I’d rather have an S-expression syntax
+    - bjorn3: Also need to think about branches and jumps, take variable amounts of arguments
+
+- Standups
+    - cfallin
+	    - I have been working on ISLE translations. Week before last, was deep in the x64 backend translating arithmetic ops into ISLE. I plan to keep pushing on that when I don’t have other cranelift related tasks as a background thing. I realized that this ISLE translation is not actually on the critical path to regalloc2, as far as I can see: 1 regalloc only accepts SSA and 2 regalloc2 has shims for non-ssa code. Can combine with the x64 mov mitosis mechanism for the 2-operand form, and actually land regalloc2 and get rid of lots of movs on the way. Several kilo-lines of code diff, but should be a perf win in compilation time and runtime
```

Comment:
s/regalloc only accepts SSA/earlier version of regalloc2 which I tried porting Cranelift to only accepted SSA/